server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -668,6 +666,23 @@ private TSExecuteStatementResp internalExecuteQueryStatement(String statement, l
     }
   }
 
+  /*
+  calculate fetch size for group by time plan
+   */
+  private int getFetchSizeForGroupByTimePlan(GroupByTimePlan groupByTimePlan) {
+    int rows =  (int) ((groupByTimePlan.getEndTime() - groupByTimePlan.getStartTime()) / groupByTimePlan
+        .getInterval());
+    // edge case for group by months when starttime is in February
+    // ie. time range = [2020-02-02, 2020-03-02), interval = 1mo, rows will get 0
```

Comment:
The comments is not good as it may cause some misleading.

rows gets 0 is caused by: the end time - the start time < the time interval.